### PR TITLE
Feat/lazy load assets

### PR DESCRIPTION
**Pull request by mingard**
In this PR:
feat: make `FieldImage` a more generic `FieldAsset`
feat: start styling `FieldAsset`
feat: introduce a `LazyLoader` container (WIP)

### CODE REVIEW

File: frontend/components/FieldAsset/FieldAsset.css [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  box-sizing: border-box;
+  border: 1px solid #e4e4e4;
+  border-radius: 5px;
+  outline: 0;
+  display: flex;
+  flex-wrap: wrap;
+  align-items: center;
+}
+
+.container[preview] {
```
Comment (eduardoboucas) [inline frontend/components/FieldAsset/FieldAsset.css]: We should only style by class name, not attribute. If you want to add a styling block based on whether a condition is met, you should use the `Style` class. See [this](https://github.com/dadi/publish/blob/master/frontend/components/Button/Button.jsx#L99) as an example.

File: frontend/containers/LazyLoader/LazyLoader.jsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import {h, Component} from 'preact'
+import {connect} from 'preact-redux'
+import proptypes from 'proptypes'
+import {bindActionCreators} from 'redux'
+import {connectHelper} from 'lib/util'
+
+import * as appActions from 'actions/appActions'
+
+class LazyLoader extends Component {
+  static propTypes = {
```
Comment (eduardoboucas) [inline frontend/containers/LazyLoader/LazyLoader.jsx]: Could you follow the same spacing convention as used in the proptypes definition of other components? (Also period at the end of the sentences, if possible)

File: frontend/containers/LazyLoader/LazyLoader.jsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    const canLoad = this.evalLoadingConditions()
+
+    if (!canLoad) {
+      if (placeholder) return (placeholder)
+
+      return null
+    }
+
+    return (<div>{children}</div>)
```
Comment (eduardoboucas) [inline frontend/containers/LazyLoader/LazyLoader.jsx]: We've been doing:

```jsx
return (
  <div>{children}</div>
)
```

... even for components that return a single element.

File: frontend/components/FieldAsset/FieldAsset.css (hunk 1 above)
Comment (mingard) [inline frontend/components/FieldAsset/FieldAsset.css]: @eduardoboucas I was just following your lead

Comment (eduardoboucas) [inline frontend/components/FieldAsset/FieldAsset.css]: I don't follow.

Comment (mingard) [inline frontend/components/FieldAsset/FieldAsset.css]: You wrote that

File: frontend/containers/LazyLoader/LazyLoader.jsx (hunk 2 above)
Comment (mingard) [inline frontend/containers/LazyLoader/LazyLoader.jsx]: Sure

File: frontend/components/FieldAsset/FieldAsset.css (hunk 1 above)
Comment (eduardoboucas) [inline frontend/components/FieldAsset/FieldAsset.css]: `multiple` is a standard HTML attribute that identifies `<select>` elements with multiple options. `preview` is something we made up.

Comment (mingard) [inline frontend/components/FieldAsset/FieldAsset.css]: > We should only style by class name, not attribute
Unless we're being consistent on this, I don't see merit in this request

File: frontend/containers/LazyLoader/LazyLoader.jsx (hunk 2 above)
Comment (mingard) [inline frontend/containers/LazyLoader/LazyLoader.jsx]: This isn't showing as resolved but as far as I can see it is!

File: frontend/components/FieldAsset/FieldAsset.css (hunk 1 above)
Comment (eduardoboucas) [inline frontend/components/FieldAsset/FieldAsset.css]: You're adding a non-standard attribute to the element, in the actual rendered DOM, without it even being prefixed with `data-`. There's no reason to do this when all you want to do is target it for styling.

That isn't the case in `FieldString`, but I can change it to also use a class if that makes you happier.

Comment (mingard) [inline frontend/components/FieldAsset/FieldAsset.css]: Yes, I think we need to be a bit more consistent here. Do that. Add to style guide too if you've the time.

Comment (eduardoboucas) [inline frontend/components/FieldAsset/FieldAsset.css]: It is in the style guide: https://github.com/dadi/publish/blob/master/docs/styling.md#21-target-class-names

Comment (eduardoboucas) [inline frontend/components/FieldAsset/FieldAsset.css]: As well as documentation for the `Style` class, the solution I proposed: https://github.com/dadi/publish/blob/master/docs/styling.md#31-the-style-class

Comment (mingard) [inline frontend/components/FieldAsset/FieldAsset.css]: Great. lets enforce it

Comment (mingard) [inline frontend/components/FieldAsset/FieldAsset.css]: I think everything else there is covered

Comment (eduardoboucas) [inline frontend/components/FieldAsset/FieldAsset.css]: I've updated `FieldString`. Can you update yours?

Comment (mingard) [inline frontend/components/FieldAsset/FieldAsset.css]: Done

**Pull request merged**